compiler/src/dotty/tools/io/JarArchive.scala

```diff
@@ -11,7 +11,7 @@ import scala.jdk.CollectionConverters.*
  * that be can used as the compiler's output directory.
  */
 class JarArchive private (root: Directory) extends PlainDirectory(root) {
-  def close(): Unit = jpath.getFileSystem().close()
+  def close(): Unit = this.synchronized(jpath.getFileSystem().close())
```

Comment:
Use reentrantlock instead？

Reply:
this is expected to be called from a single thread, is there another potential issue?

Reply:
Maybe OK, as we are not using virtual thread in JDK21 anyway.